### ISSUE TITLE
Learning rate and batch size

### ISSUE THREAD

**Issue by wzdsh** [NVIDIA/DIGITS#227]
When training with multiple GPUs, we usually use larger batch size, do we also need to increase the learning rate in this case?

**Comment by lukeyeager** [comment]
Quoting from "[One weird trick for parallelizing convolutional neural networks](http://arxiv.org/abs/1404.5997)" by Alex Krizhevsky:

> Theory suggests that when multiplying the batch size by k, one should multiply the learning rate by sqrt(k) to keep the variance in the gradient expectation constant.
...
Theory aside, for the batch sizes considered in this note, the heuristic that I found to work the best was to multiply the learning rate by k when multiplying the batch size by k. I can’t explain this discrepancy between theory and practice.

**Comment by lukeyeager** [comment]
@wzdsh, did that answer your question? If so, can you close this issue?

**Comment by wzdsh** [comment]
Thanks very much for this answer!
I have tried both multiply and not multiply the learning rate by k, the final results are quite similar.

**Comment by pliptor** [comment]
@lukeyeager What is the typical range values for k that you had and do you know what optimization algorithm and initial learning rates used?
